#156 create immunization info

KarbovskyiV opened the pull request.

KarbovskyiV committed:
156 feat: Add information block, add validations,
refactor: move formatting moethod to separate service class,
remove unused phone methods from FormTrait class,
chore: add laravel-debugbar,
add docker related files and folders to gitignore

KarbovskyiV committed:
chore: composer update, add name, avoid unbound version constraints for laravel/fortify

KarbovskyiV committed:
#64 feat: change sequence of patient search,
fix: incorrect route was used,
chore: remove omit parentheses

Vitaliy-1 commented on .env.example:
```diff
@@ -69,3 +69,8 @@ CIPHER_API_URL=
 TEST_CLIENT_ID=
 TEST_CLIENT_SECRET=
+
+WWWUSER=1000
```
Ці налаштування в dev не потрібні.

Vitaliy-1 commented on composer.json:
```diff
@@ -8,14 +9,15 @@
         "spatie/laravel-permission": "^6.16"
     },
     "require-dev": {
+        "barryvdh/laravel-debugbar": "^3.15",
```
Пропонуєте використовувати?

Vitaliy-1 commented on composer.json:
```diff
@@ -1,4 +1,5 @@
 {
+    "name": "open-health/mis",
```
Імʼя не потрібно

KarbovskyiV replied:
composer audit повернув ворнінг, то й вирішив додати, окей, видалю

Vitaliy-1 commented on app/Livewire/Encounter/Forms/Encounter.php:
```diff
@@ -112,61 +115,80 @@ class Encounter extends Form
-    ])]
     public array $immunizations;
 
+    protected function rules(): array
```
Слід використовувати або анотацію Validate або метод rules() в одному класі/формі

Vitaliy-1 commented on app/Livewire/Encounter/Forms/Encounter.php:
```diff
@@ -265,4 +315,60 @@ private function addAllowedRule(array &$rules, string $ruleKey, array $allowedVa
+     *
+     * @return RequiredIf
+     */
+    private function requiredIfPrimarySourceAndNotGivenFalse(): RequiredIf
```
Можна один метод замість 3х, який приймає 2 аргумента boolean

Vitaliy-1 commented on app/Livewire/Encounter/EncounterCreate.php:
```diff
@@ -193,31 +197,47 @@ public function searchICD10(string $value): void
-    public function save(): void
+    public function save(EncounterService $service): void
     {
         // update or create
```
А чи не було б краще для update та create створити 2 окремих компоненти? У вас все одно 2 різні routes. Таким чином було б легче читати код. Form можна одну. А для уникнення дублікації коду створити parent клас

KarbovskyiV replied:
в одному із наступних PR розділю

Vitaliy-1 commented on app/Livewire/Encounter/EncounterCreate.php:
```diff
@@ -193,31 +197,47 @@ public function searchICD10(string $value): void
-     * @throws ValidationException
      */
-    public function save(): void
+    public function save(EncounterService $service): void
```
Тут і в інших місцях сервіс краще пропускати через конструктор контролера або аналог.

Vitaliy-1 reviewed (changes requested):
В цілому, ідея із двома типами базами даних:
```php
class EncounterSQLModel implements EncouterInterface
{
   public function foo()
   {
   }
} 
```
```php
class EncounterMongoModel implements EncounterInterface
{
  public function foo()
  {
  }
}
```
```php
class EncounterRepository
{
  public function __construct(protected EncounterInterface $model)
  {
  }

  public function doSomething()
  {
      $this->model->foo()
  }
}
``` 
Сюди ж можна спробувати додати серіалізацію, яка в майбутньому інтегрується із Symphony serializer. Гадаю, не треба створювати `EncounterService`

KarbovskyiV committed:
#156 refactor: use repository instead of service for formatting data,
create ServiceProvider that based on config decide which db models to use

KarbovskyiV committed:
#156 refactor: improve requiredIf method,
move validaiton to rules method instead of validate annotation,
fix style in button-sync
